### PR TITLE
559 auth stuff

### PR DESCRIPTION
**Pull request by MAttila42**
### Summary
🎨🐛🧹

This pull request adds offline mode, retry logic, and password handling features to the frontend app. It also fixes a bug in the `RecipeFooter.vue` component and cleans up some unused style tags. The changes affect the `index.vue`, `recipe.ts`, `user.ts`, `image.ts`, and `RecipeFooter.vue` files, as well as the `.editorconfig` file.

> _We're fixing bugs and cleaning code_
> _We're adding features to our mode_
> _We're retrying requests with our token_
> _We're `uploadImage` and we're not broken_

### Walkthrough
*  Add retry logic to image and recipe store actions to handle expired or invalid user tokens ([link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/581/files?diff=unified&w=0#diff-1fc7a050bc25532b502b63820b4eb45b35dff2e4e5886aec9294fac555e37be2L33-R61), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/581/files?diff=unified&w=0#diff-fff9d1299723af9e15255bb8947c21b5d5453a9ef83d63926b63ae8b8f456b2eL127-R166), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/581/files?diff=unified&w=0#diff-fff9d1299723af9e15255bb8947c21b5d5453a9ef83d63926b63ae8b8f456b2eL226-R276), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/581/files?diff=unified&w=0#diff-fff9d1299723af9e15255bb8947c21b5d5453a9ef83d63926b63ae8b8f456b2eL253-R336))
*  Add password property to user store and store it when logging in to enable retrying login requests ([link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/581/files?diff=unified&w=0#diff-99a55108f8fc7f5654915d714ec99471d2ff4c429feecaec77d02010f186d877R4), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/581/files?diff=unified&w=0#diff-99a55108f8fc7f5654915d714ec99471d2ff4c429feecaec77d02010f186d877L51-R55))
*  Add skip login feature to index page and user store to allow offline mode ([link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/581/files?diff=unified&w=0#diff-ade85aa68406c565b477a1ae17db8373bfbdacc26ce7185dfe997711e81b213eR7-R24), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/581/files?diff=unified&w=0#diff-ade85aa68406c565b477a1ae17db8373bfbdacc26ce7185dfe997711e81b213eR46-R50))
*  Fix casing of orangeIcon attribute in RecipeFooter component to match custom directive definition ([link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/581/files?diff=unified&w=0#diff-4100b162e90ddbb8951ec90b956b8d8226eed6d822588d7c633e952fb851badbL57-R57))
*  Lower bottom margin of sign in and register links div on index page for better UI ([link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/581/files?diff=unified&w=0#diff-ade85aa68406c565b477a1ae17db8373bfbdacc26ce7185dfe997711e81b213eL20-R38))
*  Remove empty style tags from home, login, and register pages for code cleanup ([link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/581/files?diff=unified&w=0#diff-d7c7cba00436b50ec11b988659397e360494f56d2f53f7b17812723e4b13ec8bL42-L43), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/581/files?diff=unified&w=0#diff-0c5eabb00b3c5ba6eaf821bc06b45da7c540b03457b2d0d259239909abcfafb6L48-L51), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/581/files?diff=unified&w=0#diff-cc559a169f4e342b18135352add8b6435b4f12b19754c4e27d6801dddf502ff6L47-L50))
*  Apply consistent indentation style and size to all files in the project for formatting ([link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/581/files?diff=unified&w=0#diff-e3ca72daa4acabd56a67a975832ee7e5d769359e630c6519f5b8fa279e192fd0L3-R3))

### CODE REVIEW

File: apps/frontend/Gump/components/recipe/RecipeFooter.vue [hunk 1]
```diff
@@ -54,7 +54,7 @@ async function saveClick() {
 <template>
   <div v-if="props.recipe" flex="~ row" mx-2 justify-between text-left font-mono text-xl>
     <div flex="~ row" items-center>
-      <div i-fa6-solid-eye orangeIcon />
+      <div orangeicon i-fa6-solid-eye />
```
Comment (Rettend) [inline apps/frontend/Gump/components/recipe/RecipeFooter.vue:57]: itt szépen elcseszted az icont (nagybetű kell legyen)

**Review by Rettend — CHANGES_REQUESTED** [review]
lehetett volna egy composable-t csinálni a for-loop és ebből, ahelyett hogy ctrl+c mindenhova..
```ts
if (i === 0 && user.current.token !== 'offline' && statusCode.value === 401) {
            await user.login({
              username: user.current.username,
              password: user.current.password,
            })
            continue
          }
```

File: apps/frontend/Gump/components/recipe/RecipeFooter.vue (hunk 1 above)
Comment (MAttila42) [reply apps/frontend/Gump/components/recipe/RecipeFooter.vue:57]: upszika, nem vettem észre hogy azt is átírta, csak izélt, hogy rossz a sorrend

**Comment by MAttila42** [comment]
> lehetett volna egy composable-t csinálni a for-loop és ebből, ahelyett hogy ctrl+c mindenhova..
> 
> ```ts
> if (i === 0 && user.current.token !== 'offline' && statusCode.value === 401) {
>             await user.login({
>               username: user.current.username,
>               password: user.current.password,
>             })
>             continue
>           }
> ```

há én azt nem tudom hogyan kell

**Review by Rettend — APPROVED** [review]

**Pull request merged**